main/src/com/google/refine/importing/ImportingUtilities.java

```diff
@@ -438,13 +439,29 @@ static public String getRelativePath(File file, File dir) {
         String location = file.getAbsolutePath().substring(dir.getAbsolutePath().length());
         return (location.startsWith(File.separator)) ? location.substring(1) : location;
     }
-    
+
+    private static String normalizeFileName(String fileName){
+        FileSystem currentFileSystem = FileSystem.getCurrent();
+        if (currentFileSystem != FileSystem.WINDOWS) return fileName;
+        String normalizedLocalName = new String();
+        // normalize the file name if the current system is windows
+        String[] path = fileName.split("/");
```

Comment:
I do not understand why you are splitting by `/` and normalizing each part independently. Since `/` is not a allowed as part of a filename on Windows, why would you want to preserve those characters? Why not replace them by `-` like the other illegal characters? It feels like this method should basically just call `FileSystem::toLegalFileName`. I also wonder if it is worth delegating to `FileSystem::toLegalFilename` regardless of the operating system: surely, other operating systems also have illegal characters for filenames, and this method probably also handles those cases, no?

Reply:
The filename pass in is part of url and the url is splited by `/`. To minimize the influence to the create project process, I just normalize the file name and its path in the url  since only windows has such illegal while linux and mac doesn't. The method `FileSystem::toLegalFilename` is just need a file name or diretory name. However, the parameter passed in contains path and file name together. That why I split it.

Reply:
ok! Could it make sense to use existing classes like `Path` or `File` to split by path separators? Intuitively there could be cases where splitting by `/` is not quite right (for instance if we had an absolute path - not sure if that can happen at this stage though).

Reply:
OK, I'll try fix it quickly.